plugins/plugin-chart-echarts/src/Gauge/transformProps.ts

```diff
@@ -0,0 +1,226 @@
+
+const setIntervalBoundsAndColors = (
+  intervals: string,
+  intervalColorIndices: string,
+  colorFn: CategoricalColorScale,
+  normalizer: number,
+): Array<[number, string]> => {
+  const intervalBoundsNonNormalized = tokenizeToNumericArray(intervals) || [];
+  const intervalColorIndicesArray = tokenizeToNumericArray(intervalColorIndices) || [];
+  const intervalBounds = intervalBoundsNonNormalized.map(bound => bound / normalizer);
+  const intervalColors = intervalColorIndicesArray.map(ind => colorFn.colors[ind - 1]);
+  const intervalBoundsAndColors = zip([intervalBounds, intervalColors]) as Array<[number, string]>;
+  return intervalBoundsAndColors;
+};
+
+export default function transformProps(chartProps: ChartProps) {
+  const { width, height, formData, queriesData } = chartProps;
+  const {
+    metric,
+    minVal,
+    maxVal,
+    title,
+    colorScheme,
+    fontSize,
+    numberFormat,
+    roundcap,
+    showSplitLine,
+    showAxisTick,
+    startAngle,
+    endAngle,
+    intervals,
+    intervalColorIndices,
+    valueName,
+    valueAnimation,
+    valueFormatter,
+  }: EchartsGaugeFormData = { ...DEFAULT_GAUGE_FORM_DATA, ...formData };
+  const data = queriesData[0].data as TimeseriesDataRecord[];
+  const numberFormatter = getNumberFormatter(numberFormat);
+  const colorFn = CategoricalColorNamespace.getScale(colorScheme as string);
+  const normalizer = (data[0][getMetricLabel(maxVal as QueryFormMetric)] as number) || 1;
+  const intervalBoundsAndColors = setIntervalBoundsAndColors(
+    intervals,
+    intervalColorIndices,
+    colorFn,
+    normalizer,
+  );
+  const transformedData = data.map(data_point => ({
+    value: data_point[getMetricLabel(metric as QueryFormMetric)] as number,
+    name: valueName,
+  }));
+
+  const formatValue = (value: number) => valueFormatter.replace('{value}', numberFormatter(value));
+
+  console.log('formData via TransformProps.ts', formData);
+  console.log('queriesData', queriesData);
+
+  let progress;
+  let splitLine;
+  let axisLine;
+  let axisLabel;
+  let axisTick;
+  let pointer;
+  let itemStyle;
+  let detail;
+
+  if (!intervalBoundsAndColors.length) {
+    progress = {
+      show: true,
+      roundCap: roundcap,
+      width: fontSize,
+    };
+    splitLine = {
+      show: showSplitLine,
+      distance: -fontSize,
+      length: fontSize,
+      lineStyle: {
+        width: fontSize / 4,
+      },
+    };
+    axisLine = {
+      roundCap: roundcap,
+      lineStyle: {
+        width: fontSize,
+      },
+    };
+    axisLabel = {
+      distance: 2 * fontSize,
+      fontSize,
+      formatter: numberFormatter,
+    };
+    axisTick = {
+      show: showAxisTick,
+      distance: -fontSize,
+      length: fontSize / 4,
+      lineStyle: {
+        width: 2,
+      },
+    };
+    pointer = {};
+    detail = {
+      valueAnimation,
+      formatter: (value: number) => formatValue(value),
+      fontSize: 1.5 * fontSize,
+    };
+    itemStyle = {};
+  } else {
```

Comment:
If the only difference is line and item style, should we just create a function to get it from interval bound setting?